### PR TITLE
Django autocomplete light

### PR DESCRIPTION
**Pull request by akatsoulas**
@johngian r?

### CODE REVIEW

File: mozillians/api/admin.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
 from django.contrib import admin
+from django import forms
```
Comment (johngian) [inline mozillians/api/admin.py]: Please move this one line above.

File: mozillians/api/admin.py [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
-import autocomplete_light
 from import_export import fields
 from import_export.admin import ExportMixin
 from import_export.resources import ModelResource
+from dal import autocomplete
```
Comment (johngian) [inline mozillians/api/admin.py]: Same here.

File: mozillians/groups/admin.py [hunk 3]
```diff
@@ -7,10 +7,10 @@
 from django.utils.html import escape
 from django.utils.safestring import mark_safe
 
-import autocomplete_light
 from import_export.admin import ExportMixin
 from import_export.fields import Field
 from import_export.resources import ModelResource
+from dal import autocomplete
```
Comment (johngian) [inline mozillians/groups/admin.py]: Same here.

File: mozillians/mozspaces/admin.py [hunk 4]
```diff
@@ -34,12 +34,21 @@ class meta:
         model = MozSpace
 
 
+class MozSpaceAutcompleteForm(forms.ModelForm):
```
Comment (johngian) [inline mozillians/mozspaces/admin.py]: Please fix typo here. This has other occurrences too.

File: mozillians/mozspaces/admin.py [hunk 5]
```diff
@@ -2,8 +2,8 @@
 from django.contrib import admin
 from django.core.urlresolvers import reverse
 
-import autocomplete_light
 from import_export.admin import ExportMixin
+from dal import autocomplete
```
Comment (johngian) [inline mozillians/mozspaces/admin.py]: Same here.

File: mozillians/mozspaces/admin.py [hunk 6]
```diff
@@ -34,12 +34,21 @@ class meta:
...
+        }
+
+
 class MozSpaceAdmin(ExportMixin, admin.ModelAdmin):
-    form = MozSpaceAdminForm
     inlines = [PhotoAdmin, KeywordAdmin]
     search_fields = ['name']
     list_display = ['name', 'city', 'country', 'coordinator_link']
-    form = autocomplete_light.modelform_factory(MozSpace)
+    form = MozSpaceAutcompleteForm
```
Comment (johngian) [inline mozillians/mozspaces/admin.py]: Same here.

File: mozillians/templates/admin/base_site.html [hunk 7]
```diff
@@ -1,17 +0,0 @@
...
-
-{% block title %}{{ title }} | {% trans 'Django site admin' %}{% endblock %}
-
-{% block extrahead %}
-{% load static from staticfiles %}
-    <script src="{% static "mozillians/js/libs/jquery-1.11.3.min.js" %}" type="text/javascript"></script>
-    {% include 'autocomplete_light/static.html' %}{% endblock %}
-{% block branding %}
-
-<h1 id="site-name">Mozillians Administration</h1>
```
Comment (johngian) [inline mozillians/templates/admin/base_site.html]: Why are you removing everything here? This also removes the branding stuff.

File: mozillians/users/admin.py [hunk 8]
```diff
@@ -13,16 +13,17 @@
 from django.forms import ValidationError
 from django.http import HttpResponseRedirect
 
-import autocomplete_light
-from celery.task.sets import TaskSet
-from functools import update_wrapper
 from import_export.admin import ExportMixin
 from import_export.fields import Field
 from import_export.resources import ModelResource
+from celery.task.sets import TaskSet
```
Comment (johngian) [inline mozillians/users/admin.py]: Same here.

File: requirements/prod.txt [hunk 9]
```diff
@@ -135,8 +135,8 @@ diff-match-patch==20121119
 django-appconf==1.0.1
 # sha256: 3rTczTYxdaVUZdxlaZwAHDHbBeUJ6KQfzPPbYcFASsY
 statsd==1.0.0
-# sha256: __xQ7GLRNdRE3p1B3reS_yObrAhI55TRe2IF5-8QTJ4
-django-autocomplete-light==1.4.13
+# sha256: QlWFxOe0eZaRJQMTgcTLT4cawHXEAoIrg6j1VgFTHQo
```
Comment (johngian) [inline requirements/prod.txt]: Please double-check that the latest ``dal`` requires no further dependencies.

File: mozillians/templates/admin/base_site.html (hunk 7 above)
Comment (akatsoulas) [inline mozillians/templates/admin/base_site.html]: For branding we can use this
https://github.com/mozilla/mozillians/pull/1325/files#diff-b1c58232be9307720f5f4d1ff97ae5eaR49

Comment (johngian) [inline mozillians/templates/admin/base_site.html]: w00t w00t
:rocket:

**Comment by johngian** [comment]
Let's make sure that we don't leak any PII in the autocomplete endpoints.

* All autocomplete URLs used in admin should check for staff permissions
* Autocomplete URLs used in multiple curators should be available to vouched users only.

**Comment by akatsoulas** [comment]
PR updated. The curators list for groups will be in a separate pr/commit.

File: mozillians/groups/urls.py [hunk 10]
```diff
@@ -33,4 +36,8 @@
         name='toggle_skill_subscription'),
     url('^skills/search/$', 'views.search',
         dict(searched_object=Skill), name='search_skills'),
+    # Django-autocomplete-light urls
+    url('group-membership-autocomplete/$',
+        login_required(autocomplete.Select2QuerySetView.as_view(model=Group)),
+        name='group-membership-autocomplete'),
```
Comment (johngian) [inline mozillians/groups/urls.py]: This URL name ``group-membership-autocomplete`` is kind of misleading. This should be named ``group-autocomplete``.

Comment (johngian) [inline mozillians/groups/urls.py]: This might also have other references.

File: mozillians/groups/urls.py [hunk 11]
```diff
@@ -33,4 +36,8 @@
         name='toggle_skill_subscription'),
     url('^skills/search/$', 'views.search',
         dict(searched_object=Skill), name='search_skills'),
+    # Django-autocomplete-light urls
+    url('group-autocomplete/$',
+        login_required(autocomplete.Select2QuerySetView.as_view(model=Group)),
```
Comment (johngian) [inline mozillians/groups/urls.py]: Here we are using the default CBV and I am not sure we are checking for permissions (eg. ``is_staff`` or ``vouched``).

**Comment by johngian** [comment]
r+wc

**Comment by johngian** [comment]
travis is failing most probably because of master. Did you rebase?

File: mozillians/groups/urls.py (hunk 11 above)
Comment (akatsoulas) [inline mozillians/groups/urls.py]: This is not a problem for groups. You can check here:
https://github.com/mozilla/mozillians/blob/master/mozillians/groups/views.py#L79

**Comment by akatsoulas** [comment]
@johngian added one more commit

**Comment by comzeradd** [comment]
@akatsoulas you can cherry-pick [this commit](https://github.com/comzeradd/mozillians/commit/557bfa77f2c570b99afaf5519667b5c56e25ffae) that includes some style modifications to autocomplete style

**Comment by johngian** [comment]
@akatsoulas When you add a new skill that doesn't exist, the string ``Create: <foo>`` appears. It's kind of misleading for the end user. Can we modify it somehow.

**Comment by akatsoulas** [comment]
closing in favor of #1334

**Pull request closed without merging**